.github/pull_request_template.md

```diff
@@ -0,0 +1,47 @@
+**Description**
+
+<!-- 
+Please include a summary of the changes and which issue is fixed or which feature it added.
+Please also include relevant motivation and context. List any dependencies that are required for this change.
+-->
+
+<!--
+If Connected to an issue, include:
+Closes # (issue)
+-->
+
+**Type of change**
+
+<!-- Choose those that fit the type of your change and remove the others -->
+
+- Bug fix (change which fixes an issue)
+- New feature (change which adds functionality)
+- Update to functionality
+- Other: <!-- add other type here, e.g. Improve Workflow--> 
+
+**Testing**
+
+<!-- Please shortly describe how you tested your code and mark all you have done after -->
+
+<!-- exclude any of the following if they do not apply -->
+- [ ] I added basic working examples (possibly in doc-comment)
+- [ ] I added tests for large (pointer representation) values
+- [ ] I triggered all possible errors in my test in every possible way
+- [ ] I included assertions in which I compare expected values with computed values
+<!-- Please add other tests if any other have been performed -->
+
+**Checklist:**
+
+<!-- This is a short summary of the things the programmer should always consider before merging-->
+
+- [ ] I have performed a self-review of my own code
+  - [ ] Design choice fits code base + good time to do the PR
+  - [ ] The added lines provide good readability (naming, comments, documentation, spelling)
+  - [ ] The code fits our style guide
+  - [ ] I included tests for all reasonable edge cases
+  - [ ] My code works as intended and no side effects occur (e.g. memory leaks)
+  - [ ] The chosen implementation is not more complex than it has to be
+  - [ ] My code is not missing any crucial feature (or if it is listed as another task)
+  - [ ] My code uses encapsulation and modularization
+  - [ ] `cargo doc` does not generate any errors
+  - [ ] I have credited related sources if needed
```

Comment:
This is very extensive. I don't know whether we always want to copy/ write and also need to read so much stuff for every pull request. The idea of making a checklist is sufficient, but it should be short and encourage to do the checks rather than being annoyingly tedious and extensive/ long, i.e. properties that can be checked via the pipeline shouldn't be included here.

Reply:
The content does not have to be copied, since it is pasted in every time for each new PR.

Except from the cargo doc I personally think that none of the above can actually be checked by the pipeline (since cargo doc is not in the pipeline yet, I left it in for now).

I oriented on the review checklist and I personally think that having it dedicated in our pipeline raises awareness for the checks we have to do in our review. I summarized the self-review process in these checksmarks.